creed/tiling_utils_test.py

```diff
@@ -460,74 +602,86 @@ def test_tma_generate_fov_list():
         )
 
-    # the fovs list defined does not contain exactly 2 FOVs
+    # the fovs list defined does not contain exactly 4 FOVs
     with pytest.raises(ValueError):
         tiling_utils.tma_generate_fov_list(
             'sample_fovs_list.json', 3, 3
         )
 
-    # trim sample_fovs_list so it only contains 2 FOVs
-    # one to define the upper-left corner, one to define the bottom-left corner
-    sample_fovs_list['fovs'] = sample_fovs_list['fovs'][:2]
-
-    # error checking, define the upper-left x-coord to be greater than the bottom-right
-    sample_fovs_list['fovs'][0]['centerPointMicrons']['x'] = 100
-    sample_fovs_list['fovs'][1]['centerPointMicrons']['x'] = 0
-
-    # resave the data
-    with open('sample_fovs_list.json', 'w') as sfl:
-        json.dump(sample_fovs_list, sfl)
-
-    # assert test fails
-    with pytest.raises(ValueError):
-        tiling_utils.tma_generate_fov_list(
-            'sample_fovs_list.json', 3, 4
-        )
-
-    # reset x values
-    sample_fovs_list['fovs'][0]['centerPointMicrons']['x'] = 0
-    sample_fovs_list['fovs'][1]['centerPointMicrons']['x'] = 100
+    # trim sample_fovs_list so it only contains 4 FOVs, one defining each corner
+    sample_fovs_list['fovs'] = sample_fovs_list['fovs'][:4]
 
-    # error checking, define the upper-left y-coord to be less than the bottom-right
-    sample_fovs_list['fovs'][0]['centerPointMicrons']['y'] = 0
-    sample_fovs_list['fovs'][1]['centerPointMicrons']['y'] = 100
-
-    # resave the data
+    # resave with just 4 FOVs
     with open('sample_fovs_list.json', 'w') as sfl:
         json.dump(sample_fovs_list, sfl)
 
-    # assert test fails
-    with pytest.raises(ValueError):
-        tiling_utils.tma_generate_fov_list(
-            'sample_fovs_list.json', 3, 4
-        )
-
-    # reset y values
-    sample_fovs_list['fovs'][0]['centerPointMicrons']['y'] = 100
-    sample_fovs_list['fovs'][1]['centerPointMicrons']['y'] = 0
-
-    # resave the data
-    with open('sample_fovs_list.json', 'w') as sfl:
-        json.dump(sample_fovs_list, sfl)
+    # NOTE: we'll save the coordinate checks for test_validate_tma_corners
 
     # create the FOV regions
-    # use 3 and 4 since 3 divides into clean ints and 4 does not
     fov_regions = tiling_utils.tma_generate_fov_list(
-        'sample_fovs_list.json', 3, 4
+        'sample_fovs_list.json', 4, 3
     )
 
-    center_points = [fov_regions[fov] for fov in fov_regions]
+    # define the generated fov_names and center points
+    fov_names = list(fov_regions.keys())
+    center_points = list(fov_regions.values())
 
     # define the actual center points created
-    # NOTE: y-coordinates are intentionally rounded
-    actual_x = [0, 50, 100]
-    actual_y = [100, 66, 33, 0]
-    actual_center_points = [
-        (x, y) for x in actual_x for y in actual_y
-    ]
-
-    # assert the centroids are the same
-    assert center_points == actual_center_points
+    if rhombus_test_name == 'rectangle':
```

Comment:
OK, I had code to do that originally but removed it. I can add it back in.

Reply:
I think it will still be useful to test with all slant patterns, however I'll remove the coordinate tests (allow `test_generate_x_y_fov_pairs_rhombus` to handle that) and simply ensure the FOVs are in the correct order.

Reply:
Added both an explicit corner FOV name check and a check to ensure that all the FOVs are in the correct order in the TMA auto name to coord `dict`.